File: web/medistream/src/features/consultation/services/consultationService.js
import API from '../../../services/api';

export const consultationService = {
    addConsultation: async (consultationData) => {
        try {
            const response = await API.post('/api/consultations/add', consultationData);
            return response.data;
        } catch (error) {
            console.error("Error saving consultation:", error);
            throw error;
        }
    },

    getAllConsultations: async () => {
        try {
            const response = await API.get('/api/consultations/all');
            return response.data;
        } catch (error) {
            console.error("Error fetching consultations:", error);
            throw error;
        }
    },

    updateConsultation: async (consultationId, consultationData) => {
        try {
            const response = await API.put(`/api/consultations/update/${consultationId}`, consultationData);
            return response.data;
        } catch (error) {
            console.error("Error updating consultation:", error);
            throw error;
        }
    },

    deleteConsultation: async (consultationId) => {
        try {
            const response = await API.delete(`/api/consultations/delete/${consultationId}`);
            return response.data;
        } catch (error) {
            console.error("Error deleting consultation:", error);
            throw error;
        }
    }
};

File: mobile/app/src/main/java/com/medistream/staff/consultations/ConsultationService.kt
package com.medistream.staff.consultations

import retrofit2.Response
import retrofit2.http.*

interface ConsultationService {
    @GET("api/consultations/all")
    suspend fun getConsultations(): Response<List<ConsultationItem>>

    @POST("api/consultations/add")
    suspend fun createConsultation(@Body request: ConsultationRequest): Response<ConsultationItem>

    @PUT("api/consultations/update/{id}")
    suspend fun updateConsultation(@Path("id") id: Int, @Body request: ConsultationRequest): Response<ConsultationItem>

    @DELETE("api/consultations/delete/{id}")
    suspend fun deleteConsultation(@Path("id") id: Int): Response<Void>

    @GET("api/consultations/patient/{patientId}")
    suspend fun getConsultationsByPatient(@Path("patientId") patientId: Int): Response<List<ConsultationItem>>
}
